src/lib.rs

```diff
@@ -224,14 +234,56 @@ impl<'a> FMatcher<'a> {
                         }
                         return Ok(());
                     } else {
+                        if self.skip_blank_lines(&mut ptn_lines, Some(x)).0.is_none() {
+                            return Ok(());
+                        }
                         return Err(text_lines_off);
                     }
                 }
-                (None, Some(_)) => return Err(text_lines_off),
+                (None, Some(x)) => {
+                    let (x, skipped) = self.skip_blank_lines(&mut text_lines, Some(x));
+                    if x.is_none() {
+                        return Ok(());
+                    }
+                    return Err(text_lines_off + skipped);
+                }
             }
         }
     }
 
+    /// If requested by the user, skip blank lines (including, optionally, the current blank line
```

Comment:
First sentence here is quite long. I'd consider breaking it down.

Reply:
This function is surprisingly complex, so it's a bit hard to document. Does the fix in https://github.com/softdevteam/fm/pull/4/commits/06411a4e309cd958738c377c0d818019968e44a6 work for you?